packages/cli/src/execution-lifecycle/execution-lifecycle-hooks.ts

```diff
@@ -192,14 +185,29 @@ function hookFunctionsPush(): IWorkflowExecuteHooks {
 	};
 }
 
-function hookFunctionsPreExecute(): IWorkflowExecuteHooks {
+function hookFunctionsExternalHooks(): IWorkflowExecuteHooks {
 	const externalHooks = Container.get(ExternalHooks);
 	return {
 		workflowExecuteBefore: [
 			async function (this: WorkflowHooks, workflow: Workflow): Promise<void> {
 				await externalHooks.run('workflow.preExecute', [workflow, this.mode]);
 			},
 		],
+		workflowExecuteAfter: [
+			async function (this: WorkflowHooks, fullRunData: IRun) {
+				await externalHooks.run('workflow.postExecute', [
+					fullRunData,
+					this.workflowData,
+					this.executionId,
+				]);
```

Comment:
This used to be wrapped in a try-catch that ignored the error. I take it now that's covered when we run the hook.

Reply:
no other external hook (besides `n8n.stop`) is wrapped in try-catch, and I also clearly remember Jan telling me that if an external hook throws, then we throw that error. If someone has an external hook that breaks because of this, then they need to catch the error in the hook code.
External hooks aren't fire-and-forget. This is also why we await on them and run them sequentially.